### PR TITLE
Tickets/dm 9940

### PR DESCRIPTION
**Pull request by isullivan**
This branch includes improvements to status messages for debugging, better dynamic convergence tests when working with real data, and removes the dependence on `Bandpass` from `sims_photUtils`. The filter profile approximation in `BandpassHelper` is of course not as accurate as using the measured filter profile from `sims_photUtils`, so the test results degraded slightly. There should be no need to compare the changes to any of the `.npy` test files.

### CODE REVIEW

File: python/buildDcrCoadd.py [hunk 1]
```diff
@@ -357,26 +365,31 @@ def build_model(self, verbose=True, max_iter=10, min_iter=None, clamp=None,
             initial_solution = self._model_spatial_filter(self.model, spatial_filter_use)
             if refine_max_iter is None:
                 refine_max_iter = max_iter*2
+            if refine_convergence_threshold is None:
+                refine_convergence_threshold = convergence_threshold/10.
```
Comment (morriscb) [inline python/buildDcrCoadd.py]: Any reasons for the magic numbers here other than just making the refined threshold an order of magnitude finer than the convergence_threshold?

File: python/buildDcrCoadd.py [hunk 2]
```diff
@@ -418,6 +436,10 @@ def _build_model_subroutine(self, initial_solution, verbose=True, max_iter=10, m
         Sets self.model as a list of np.ndarrays
         Sets self.weights as a np.ndarray
         """
+        if divergence_threshold is None:
+            divergence_threshold = 1.05
+        if obs_divergence_threshold is None:
+            obs_divergence_threshold = 1.05
```
Comment (morriscb) [inline python/buildDcrCoadd.py]: Maybe something a brief explanation of these numbers would be in order? As in why 1.05 and not 1.01?

File: python/buildDcrCoadd.py [hunk 3]
```diff
@@ -486,7 +510,12 @@ def _build_model_subroutine(self, initial_solution, verbose=True, max_iter=10, m
                      np.sum(np.abs([soln[inds_use] for soln in last_solution])))
             if verbose:
                 print("Iteration %i: delta=%f" % (sol_iter, delta))
-            if delta > last_delta:
+                print("Convergence-weighted gain used: %f" % gain)
+            if n_exp_cut != n_exp_cut_last:
+                divergence_threshold_use = 2.0
```
Comment (morriscb) [inline python/buildDcrCoadd.py]: A brief explanation of the number here would be nice.

File: python/dcr_utils.py [hunk 4]
```diff
@@ -126,9 +220,9 @@ def fft_shift_convolve(image, shift, n_substep=100, useInverse=False, weights=No
     """
     y_size, x_size = image.shape
     kernel_x = kernel_1d(shift.dx, x_size, n_substep=n_substep, debug_sinc=False,
-                         useInverse=useInverse, weights=weights)
+                         useInverse=useInverse, weights=weights, lanczos=4)
```
Comment (morriscb) [inline python/dcr_utils.py]: Why was 4 found to be appropriate order and do you want to make this a defined constant elsewhere?

File: python/dcr_utils.py [hunk 5]
```diff
@@ -126,9 +220,9 @@ def fft_shift_convolve(image, shift, n_substep=100, useInverse=False, weights=No
...
     kernel_y = kernel_1d(shift.dy, y_size, n_substep=n_substep, debug_sinc=False,
-                         useInverse=useInverse, weights=weights)
+                         useInverse=useInverse, weights=weights, lanczos=4)
```
Comment (morriscb) [inline python/dcr_utils.py]: Same with this 4.

File: tests/testDcrModelGeneration.py [hunk 6]
```diff
@@ -132,15 +133,15 @@ def test_calculate_new_model(self):
...
-            self.assertFloatsAlmostEqual(inv_var_new, inv_var_ref)
+            self.assertFloatsAlmostEqual(inv_var_new, inv_var_ref, printFailures=self.printFailures)
 
     def test_calc_model_metric(self):
         """Test that the DCR model convergence metric is calculated consistently."""
         model_file = "test_data/build_model_vals.npy"
-        metric_ref = np.array([0.022835006051, 0.014885210227, 0.0120010522772,
-                               0.0140386530451, 0.0204993350076, 0.0237343323628])
+        metric_ref = np.array([0.0719102938278, 0.0243621751, 0.00996798332164,
+                               0.0235583355852, 0.0309890648232, 0.0339528991862])
```
Comment (morriscb) [inline tests/testDcrModelGeneration.py]: Maybe a short description of where these values are coming from would be in order. As in "This is the expected result from running this part of the DRC pipeline in [this mode]".

File: python/generateTemplate.py [hunk 7]
```diff
@@ -567,87 +560,13 @@ def load_bandpass(filter_name='g', wavelength_step=None, use_mirror=True, use_le
...
-            component_list += lens_list
-        if use_atmos:
-            component_list += atmos_list
-        if use_detector:
-            component_list += detector_list
-        if use_filter:
-            component_list += filter_list
-        bandpass.readThroughputList(rootDir=throughput_dir, componentList=component_list)
-        # Calculate bandpass phi value if required.
-        if bandpass.phi is None:
```
Comment (morriscb) [inline python/generateTemplate.py]: I take it this large amount of code was moved to BandPassHelper?

Comment (isullivan) [reply python/generateTemplate.py]: Correct.

**Comment by morriscb** [comment]
Looks good and ready to merge assuming it passes a linter run.

**Pull request merged**